### ISSUE TITLE
Unclear what "Protocol Binding document" refers to in section 8.3

### ISSUE THREAD

**Issue by benfrancis** [w3c/wot-thing-description#1326]
In section 8.3 Protocol Bindings there is a note which says:

> Optimally, the protocols used are listed as a scheme in the IANA registry [IANA-URI-SCHEMES]). This guarantees a unique Protocol Binding assignment. In case the desired protocol is not yet registered with IANA, it is recommended to follow the scheme value of the protocol specifications, if available. In principle, to avoid ambiguity in the identification of the protocol via the scheme, the Protocol Binding document will provide a recommended scheme value to enable unique protocol identification in the context of WoT. 

I understand that this note is meant to [replace a normative assertion from the WoT Architecture specification](https://github.com/w3c/wot-architecture/issues/639) which strictly constrained valid WoT protocols to those with URI schemes registered with IANA.

I commented on an earlier draft of this note in https://github.com/w3c/wot-thing-description/pull/1301#issuecomment-979352573 regarding my concerns about the loophole it introduces in allowing any non-web (e.g. MQTT) or non-internet (e.g. Zigbee) protocol to be used as a WoT protocol (see also: https://github.com/w3c/wot-binding-templates/issues/137) by using made-up/unofficial URI schemes.

I appreciate that this latest text attempts to narrow the set of possible URI schemes further by stating that "the Protocol Binding document will provide a recommended scheme", but it's not clear what is meant by "protocol binding document". Does it mean the [WoT Binding Templates](https://w3c.github.io/wot-binding-templates) Working Group Note, or an individual protocol binding template sub-specification?

If it's the former then it shifts the responsibility for accepting new protocols from IANA to the editors of the WoT Binding Template specification, with the table in [section 4.1](https://w3c.github.io/wot-binding-templates/#protocol-intro) acting as a kind of registry.

If it's the latter then anyone can make up their own URI scheme using a TD context extension as defined in [section 7.2](https://w3c.github.io/wot-thing-description/#adding-protocol-bindings).

Either way I want to note that this amounts to a significant dilution of the previous normative assertion because:
1. It opens to the door to any non-web and non-internet protocol being used in a Thing Description, drastically increasing the number of potential protocols a WoT Consumer may be expected to support
2. Neither this note, nor the WoT Binding Template note, nor individual protocol binding template documents are normative

I understand that this is a conscious decision on the part of the Editors in order to accommodate popular protocols like MQTT and Modbus, and I respect that decision. I just want to make sure that people understand the significance of the change, and ideally make the precise intention of the note clearer.

**Comment by egekorkan** [comment]
Call of 15.12.2021:
We can fix this by constraining the list to the ones specified by binding template + IANA registry. It is not clear whether we can refer to the Binding Templates which is an informative document. If not possible, there is a possibility to make the binding document normative in the next charter.

**Comment by relu91** [comment]
For the record, this issue is related to https://github.com/w3c/wot-thing-description/issues/1321. In [my comment](https://github.com/w3c/wot-thing-description/issues/1321#issuecomment-992661716) I was suggesting sections that should be normative to solve this issue.

**Comment by benfrancis** [comment]
To be clear, would the list of protocols be limited to those listed in the table in [section 4.1](https://w3c.github.io/wot-binding-templates/#protocol-intro) of the WoT Binding Templates document (in order to guarantee a unique mapping with URI schemes), or just any individual binding template document which anyone can create and reference via a context extension?

@relu91 I don't think WoT Binding Templates can be made normative under the current charter 😟

**Comment by egekorkan** [comment]
I think I have forgotten to add the proposal from the TD call. We want to say that only the protocols defined by the binding templates can be used in the TD spec. We can also allow IANA registered protocols but if they cannot be mapped to WoT operations or do not have a vocabulary, it is not very helpful. The tricky part is that Binding Templates is not a normative spec so just referencing to that **may** not be enough. Worst case, we can copy over the table of currently specified protocols in the binding to the TD spec

**Comment by mlagally** [comment]
> We want to say that only the protocols defined by the binding templates can be used in the TD spec. We can also allow IANA registered protocols but if they cannot be mapped to WoT operations or do not have a vocabulary, it is not very helpful. The tricky part is that Binding Templates is not a normative spec so just referencing to that **may** not be enough. Worst case, we can copy over the table of currently specified protocols in the binding to the TD spec

I think this would be a significant change in policy and is a very different interpretation of the normative language of the architecture spec, which says: Only use protocols where you have a URL. 

I don't think it is a wise choice to constraint the protocols that can be used with a Thing description.
If the TD task force would consider to go into that direction, it needs significant discussions between task forces about the benefits and adoption risks. 

If there would be consensus on a restricted list of supported protocols, these constraints must be in the TD specification itself. The binding templates are informative documents.

**Comment by egekorkan** [comment]
> I think this would be a significant change in policy and is a very different interpretation of the normative language of the architecture spec, which says: Only use protocols where you have a URL.

This is not correct. All protocols defined by the binding templates have and should have a URI scheme definition.

**Comment by egekorkan** [comment]
@benfrancis please see #1342

**Comment by benfrancis** [comment]
I note that #1342 was closed. Has there been any further discussion around this?
